docs/streams/developer-guide/dsl-api.html

```diff
@@ -3097,152 +3098,994 @@ <h5><a class="toc-backref" href="#id34">KTable-KTable Foreign-Key
+            if (record.value() == null) {
+                return; // Skip null values
+            }
+
+            // Replace slang words in the message
+            final String[] words = record.value().split(&quot;\\s+&quot;);
+            for (final String word : words) {
+                String replacedWord = SLANG_DICTIONARY.getOrDefault(word, word);
+                context().forward(record.withValue(replacedWord));
+            }
+        }
+    }
+}</code></pre>
+                <h4>Keynotes</h4>
+                <ul>
+                    <li><strong>Type Safety and Flexibility:</strong> The process and processValues APIs utilize
+                        <code>ProcessorContext</code> and <code>Record</code> or <code>FixedKeyRecord</code> objects for better type
+                        safety and flexibility of custom processing logic.
+                    </li>
+                    <li><strong>Clear State and Logic Management:</strong> Implementations for <code>Processor</code> or
+                        <code>FixedKeyProcessor</code> should manage state and logic clearly. Use <code>context().forward()</code>
+                        for emitting records downstream.
+                    </li>
+                    <li><strong>Unified API:</strong> Consolidates multiple methods into a single, versatile API.</li>
+                    <li><strong>Future-Proof:</strong> Ensures compatibility with the latest Kafka Streams releases.</li>
+                </ul>
+            </div>
+            <div class="section" id="transformers-removal-and-migration-to-processors">
+                <a class="headerlink" href="#transformers-removal-and-migration-to-processors" title="Permalink to this headline">
+                    <h3>
+                        <a class="toc-backref" href="#id37">Transformers removal and migration to processors</a>
+                    </h3>
+                </a>
+                <p>As of Kafka 4.0, several deprecated methods in the Kafka Streams API, such as <code>transform</code>,
+                    <code>flatTransform</code>,<br><code>transformValues</code>, <code>flatTransformValues</code>, and
+                    <code>process</code> have been removed. These methods have been replaced with the<br>more versatile Processor API.
+                    This guide provides detailed steps for migrating existing code to use the new Processor<br>API and explains the
+                    benefits of the changes.</p>
+                <p>The following deprecated methods are no longer available in Kafka Streams:</p>
+                <ul>
+                    <li><code>KStream#transform</code></li>
+                    <li><code>KStream#flatTransform</code></li>
+                    <li><code>KStream#transformValues</code></li>
+                    <li><code>KStream#flatTransformValues</code></li>
+                    <li><code>KStream#process</code></li>
```

Comment:
Should we somehow highlight that we refer to the "old" `process(...)` methods -- there is still the new `process()` method -- otherwise, could be confusing?